### PR TITLE
Adding the ListDirectedGraphs class

### PR DESCRIPTION
**Pull request by uemarica**
I added a class which has methods to convert a given directed graph to its corresponding line graph.

### CODE REVIEW

**Comment by jkinable** [comment]
Thanks. Before we can have a look at it:

- please copy&paste the header from another class into your 2 new files. ONLY change the year and your name, nothing else
- please format your javadoc properly. Make sure it is properly indented. 
- add some more tests
- the package name 'own' doesn't seem to be very appropriate.
- the algorithm should work for any type of vertices/edges, not just for <Integer,DefaultEdge>
- remove commented out code and unused imports

File: jgrapht-core/src/test/java/org/jgrapht/LineDirectedGraphsTest.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import org.junit.Test;
+
+import java.util.Set;
+
+import static org.junit.Assert.*;
+/* Test for line directed graph
+* @author Arica Chakraborty*/
+
+public class LineDirectedGraphsTest {
+    /*Test to check whether the createListGraph method of the ListDirectedGraphs class is working for diorected graphs or not*/
```
Comment (tibrewalpratik17) [inline jgrapht-core/src/test/java/org/jgrapht/LineDirectedGraphsTest.java]: @uemarica I think there is a typo here

File: jgrapht-core/src/main/java/org/jgrapht/LineDirectedGraphs.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    public static <V, E> boolean isEmpty(Graph<V, E> graph)
+    {
+        Objects.requireNonNull(graph, "Graph can't be null");
+        return graph.edgeSet().isEmpty();
+    }
+    SimpleDirectedGraph resultGraph;
+    /*Method that creates a line graph out of the given input graph
+    * @param graph is a graph
+    *@param <V> the graph vertex type
```
Comment (tibrewalpratik17) [inline jgrapht-core/src/main/java/org/jgrapht/LineDirectedGraphs.java]: There will be a space before `@param`

File: jgrapht-core/src/main/java/org/jgrapht/LineDirectedGraphs.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    * @param i2 is an Vertex Object
+    * @return boolean value*/
+
+    public <V,E>boolean isConnected(V i1,V i2){
+        boolean answer = false;
+        int rem,div;
+        Set<V> setVertex = resultGraph.vertexSet();
+        if(setVertex.contains(i1) && setVertex.contains(i2)) {
+
+
```
Comment (tibrewalpratik17) [inline jgrapht-core/src/main/java/org/jgrapht/LineDirectedGraphs.java]: I think you can remove this extra line.

File: jgrapht-core/src/test/java/org/jgrapht/LineDirectedGraphsTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import org.jgrapht.Graph;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.SimpleDirectedGraph;
+import org.junit.Test;
+
+import java.util.Set;
+
+import static org.junit.Assert.*;
+/* Test for line directed graph
+* @author Arica Chakraborty*/
```
Comment (tibrewalpratik17) [inline jgrapht-core/src/test/java/org/jgrapht/LineDirectedGraphsTest.java]: Indentation error

File: jgrapht-core/src/main/java/org/jgrapht/LineDirectedGraphs.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+ * the Eclipse Foundation.
+ */
+/*Conversion of a given graph to its line graph
+* This algorithm returns back a directed line graph for the input directed graph*/
+
+public class LineDirectedGraphs {
+    /*Method that checks whether the given graph is Empty or not
+    *  @param graph is a graph
+    *  @param <V> the graph vertex type
+     * @param <E> the graph edge type
```
Comment (tibrewalpratik17) [inline jgrapht-core/src/main/java/org/jgrapht/LineDirectedGraphs.java]: Improper indentation.

**Comment by jkinable** [comment]
We reworked out [Developer Wiki](https://github.com/jgrapht/jgrapht/wiki#developer-pages) Please have a look at the various articles. These provide much more in depth guidelines for your first submission. This will significantly shorten the time it takes to get your submission accepted.

**Comment by jkinable** [comment]
@tibrewalpratik17 could you make the requested modifications such that we can finish this PR?

**Pull request closed without merging**